src/main/java/com/roleManagement/rolebasedManagement/controller/ActionController.java

```diff
@@ -15,15 +17,24 @@ public class ActionController {
     Logger logger= LoggerFactory.getLogger(ActionController.class);
     @Autowired
     ActionService actionService;
+    @Autowired
+    ActionRepository actionRepository;
     @GetMapping("/actions")
-    public List<Action> getAllAction(){
+    public List<Action> getAllAction()  {
         logger.info("Getting all the Action list");
          return actionService.listAllAction();
     }
 
 @PostMapping("/saveAction")
     public Action add(@RequestBody @Valid Action action){
-        logger.info("Adding action to the db");
-            return actionService.saveAction(action);
+    logger.info("Adding role to the db ");
+    if(this.actionRepository.existsByActionName(action.getActionName())){
+        logger.error("Data is present in the db");
+        throw new DuplicateResource("Data is alredy there in db");
```

Comment:
Not proper handling of Exception. It is asked to send only error message & error code to the user.